cni/pkg/ambient/net.go

```diff
@@ -707,8 +798,118 @@ func (s *Server) CreateRulesOnNode(ztunnelVeth, ztunnelIP string, captureDNS boo
 	return nil
 }
 
+func (s *Server) updateZtunnelEbpfOnNode(pod *corev1.Pod, captureDNS bool) error {
+	if s.ebpfServer == nil {
+		return fmt.Errorf("uninitialized ebpf server")
+	}
+
+	ip := pod.Status.PodIP
+	args, err := buildEbpfParaByIP(ip, true, false)
+	if err != nil {
+		return err
+	}
+	args.CaptureDNS = captureDNS
+
+	log.Infof("update ztunnel ebpf args: %+v", args)
+	s.ebpfServer.AcceptRequest(args)
+	return nil
+}
+
+func (s *Server) delZtunnelEbpfOnNode() error {
+	if s.ebpfServer == nil {
+		return fmt.Errorf("uninitialized ebpf server")
+	}
+
+	args := &ebpf.RedirectArgs{
+		Ifindex:   0,
+		IsZtunnel: true,
+		Remove:    true,
+	}
+	log.Infof("del ztunnel ebpf args: %+v", args)
+	s.ebpfServer.AcceptRequest(args)
+	return nil
+}
+
+func (s *Server) updatePodEbpfOnNode(pod *corev1.Pod, ip string) error {
+	if s.ebpfServer == nil {
+		return fmt.Errorf("uninitialized ebpf server")
+	}
+
+	if ip == "" {
+		ip = pod.Status.PodIP
+	}
+	if ip == "" {
+		log.Debugf("skip adding pod %s/%s, IP not yet allocated", pod.Name, pod.Namespace)
+		return nil
+	}
+
+	args, err := buildEbpfParaByIP(ip, false, false)
+	if err != nil {
+		return err
+	}
+
+	log.Infof("update POD ebpf args: %+v", args)
+	s.ebpfServer.AcceptRequest(args)
+	return nil
+}
+
+func (s *Server) AddPodToMesh(pod *corev1.Pod, ip string) {
+	switch s.redirectMode {
+	case IptablesMode:
+		addPodToMeshWithIptables(pod, ip)
+	case EbpfMode:
+		if err := s.updatePodEbpfOnNode(pod, ip); err != nil {
+			log.Errorf("failed to update POD ebpf: %v", err)
+		}
+	}
+}
+
+func (s *Server) delPodEbpfOnNode(ip string) error {
+	if s.ebpfServer == nil {
+		return fmt.Errorf("uninitialized ebpf server")
+	}
+
+	ipAddr, err := netip.ParseAddr(ip)
+	if err != nil {
+		return fmt.Errorf("failed to parse ip(%s): %v", ip, err)
+	}
+
+	ifIndex := 0
+
+	if veth, err := getVethWithDestinationOf(ip); err != nil {
+		log.Infof("failed to get device: %v", err) // Change to debug, it's expected intf has been destroyed
```

Comment:
So does we expect getVethWithDestinationOf would always be 0? Otherwise, why is L877 safe?

Reply:
We try to get the ifIndex by Ipaddress , if failed, we will set it to 0, and it’s safe to call the following API.

Reply:
Why is it safe to use zero by default? If its always safe to use 0, why not always use it?

Reply:
THis is to handle pod deletion event.  In this case, veth has already been removed by CNI, so ifindex does not matter -- we just need to clean up the appinfo map in this case. 

2nd case is pod is not deleted but only remove out of ambient mode, in this case, ifindex matters.